moveit_core/robot_state/include/moveit/robot_state/robot_state.h

```diff
@@ -1770,6 +1812,19 @@ as the new values that correspond to the group */
    */
   double testJointSpaceJump(const JointModelGroup* group, std::vector<RobotStatePtr>& traj, double jump_threshold);
 
+  /**
+   * \brief Tests for absolute joint space jumps of a trajectory. The absolute difference in joint space between
+   * sequential points for each active joint is computed and if it exceededs \e prismatic_jump_threshold for prismatic
+   * joints or revolute_jump_threshold for revolute joints, the returned path is truncated up to just before the jump.
+   * @param group The joint model group of the robot state.
+   * @param traj The trajectory that should be tested.
+   * @param prismatic_jump_threshold The threshold to determine if a joint space jump has occurred .
+   * @param revolute_jump_threshold The threshold to determine if a joint space jump has occurred .
+   * @return The fraction of the trajectory that passed.
+   */
+  double testJointSpaceJump(const JointModelGroup* group, std::vector<RobotStatePtr>& traj,
```

Comment:
make this function private

Reply:
If these functions are private, I think it suffices to have a the struct-version only.

Reply:
Unless you feel strongly that these three should be combined, I'm inclined to leave it. Having the subfunctions makes the code a lot cleaner IMHO.